TWEAK: Merc kits and some fixes

LordNest opened the pull request:
ПР делает что-то умное

close #1550
close #1527
close #1523
close #1521
close #1470
close #1412
close #1330


### Чейнджлог
```yml
🆑LordNest
bugfix: Починил цветам свет. Но это не точно.
rscadd: Вернул из небытия наборы наёмников. Только будьте осторожны. Они теперь в ящиках, ибо юнит тесты
maptweak: Поставил инфотехнику отдельный новый прикольный шкаф
balance: Добавил наёмникам набор нетраннера. Использовать на свой страх и риск.
admin: Включил добавленное уже сто лет назад Respawn as Human. Теперь оно Respawn as Self
/🆑
```

<!--
  Честно заполняем галочки. Чем больше галочек, тем быстрее проверять Pull Request, соответственно он быстрее будет принят.
  Чтобы отметить - ставим `x` (икс) внутри квадратных скобочек вот так: `- [x] ...`.
  Галочки можно доставлять позже по мере окончания работы над PR'ом.
-->

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.

LordNest committed:
TWEAK: Merks kits

LordNest committed:
Эквип Южика

LordNest committed:
Вижу цель, не вижу препятствий

LordNest committed:
оно всё равно не использовалось

LordNest committed:
Фиксы, фиксы, сука, фиксы

LordNest committed:
проходим тесты мерками

LordNest committed:
забыл ящик нормально сделать

LordNest committed:
забыл, сейчас вспомнил

SuhEugene commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -144,6 +144,9 @@ var/global/list/admin_verbs_spawn = list(
 	/datum/admins/proc/spawn_atom,		// allows us to spawn instances,
 	/datum/admins/proc/spawn_artifact,
 	/client/proc/spawn_chemdisp_cartridge,
+	// [SIERRA-ADD] - admin_verbs
+	/client/proc/respawn_as_self,
+	// [/SIERRA-ADD]
```
Воооот бы это на ооооффы на саамом деееле. Мечты-мечты...

```suggestion
	// [SIERRA-ADD] - ADMIN_VERBS
	/client/proc/respawn_as_self,
	// [/SIERRA-ADD]
```

SuhEugene replied:
А ещё это в моде `CLIENT_VERBS` находится уже
Называется `/client/proc/respawn_as_human` - `Respawn as Human`

```suggestion
	// [SIERRA-ADD] - CLIENT_VERBS
	/client/proc/respawn_as_human,
	// [/SIERRA-ADD]
```

SuhEugene commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -21,6 +21,23 @@ This saves us from having to call add_fingerprint() any time something is put in
 				update_inv_l_hand(0)
 			else
 				update_inv_r_hand(0)
+			// [SIERRA-ADD]
```
Прилепим к моду ССИнпута по причине ну это же для того чтобы на `E` работало круто. И туда же в README.md надо это дописать.
```suggestion
			// [SIERRA-ADD] - SSINPUT
```

SuhEugene commented on maps/sierra/items/modular_computer.dm:
```diff
@@ -174,3 +174,24 @@
 /obj/item/modular_computer/pda/wrist/lila/black
 	icon_state = "wrist-lila-black"
 	icon_state_unpowered = "wrist-lila-black"
+
+// Laptop icon override
+
+/obj/item/modular_computer/laptop
+	icon = 'maps/sierra/icons/obj/modular_laptop.dmi'
+
+/obj/item/modular_computer/laptop/verb/rotatelaptop()
+	set name = "Rotate laptop"
+	set category = "Object"
+	set src in view(1)
+
+	if(usr.stat == DEAD)
+		if(!round_is_spooky())
+			to_chat(src, "<span class='warning'>The veil is not thin enough for you to do that.</span>")
+			return
+
+	src.set_dir(turn(src.dir, -90))
+
+/obj/item/modular_computer/laptop/update_verbs()
+	..()
+	verbs |= /obj/item/modular_computer/laptop/verb/rotatelaptop
```
Оффы не хотят, да?

LordNest replied:
> Оффы не хотят, да?

Ну я давно уже спрашивал, они говорили, что иконки не те. Мб потом как-нибудь

SuhEugene commented on mods/admin_verbs/README.md:
```diff
@@ -0,0 +1,79 @@
+
+#### Список PRов:
+
+- https://github.com/SierraBay/SierraBay12/pull/#####
```
```suggestion
- https://github.com/SierraBay/SierraBay12/pull/1798
```

SuhEugene commented on mods/admin_verbs/README.md:
```diff
@@ -0,0 +1,79 @@
+### Оверрайды
+
+- `mods/_master_files/sound/my_cool_sound.ogg`
+- `mods/_master_files/code/my_modular_override.dm`: `proc/overriden_proc`, `var/overriden_var`
```
```suggestion
- Отсутствуют
```

SuhEugene commented on mods/admin_verbs/README.md:
```diff
@@ -0,0 +1,79 @@
+
+### Дефайны
+
+- `code/__defines/~mods/example.dm`: `EXAMPLE_SPEED_MULTIPLIER`, `EXAMPLE_SPEED_BASE`
```
```suggestion
- Отсутствуют
```

SuhEugene commented on mods/admin_verbs/README.md:
```diff
@@ -0,0 +1,79 @@
+
+### Используемые файлы, не содержащиеся в модпаке
+
+- `mods/_master_files/icons/obj/alien.dmi`
```
```suggestion
- Отсутствуют
```

SuhEugene commented on mods/admin_verbs/_admin_verbs.dm:
Мод бракую полностью.

SuhEugene commented on mods/fancy_sofas/_fancy_sofas.dme:
```diff
@@ -7,3 +7,12 @@
 #include "code/sofa.dm"
 
 #endif
+// BEGIN_INTERNALS
+// END_INTERNALS
+// BEGIN_FILE_DIR
+#define FILE_DIR .
+// END_FILE_DIR
+// BEGIN_PREFERENCES
+// END_PREFERENCES
+// BEGIN_INCLUDE
+// END_INCLUDE
```
```suggestion
```

SuhEugene reviewed (changes requested):
Мод на Respawn as Character бракую полностью. Уже было это добавлено Каддлом в мод `CLIENT_VERBS`, прок называется `/client/proc/respawn_as_human` - `Respawn as Human`. Он просто не был добавлен в список админ-вёрбов.

Остальное там по мелочи.

SuhEugene commented on maps/sierra/structures/other.dm:
```diff
@@ -10,3 +10,137 @@
+			virtualIcon.DrawBox(null, B.b1x1, B.b1y,  B.b1x2, B.b1y) // horizontal line, left to right
+			virtualIcon.DrawBox(null, B.b2x, B.b2y1,  B.b2x, B.b2y2) // vertical line, top to bottom
+
+		overlays += bmark // add the decal
```
```suggestion
		AddOverlays(bmark) // add the decal
```

SuhEugene commented on maps/sierra/structures/other.dm:
```diff
@@ -10,3 +10,137 @@
+	if(isWelder(W))
+		var/obj/item/weldingtool/WT = W
+		if(WT.remove_fuel(0, user))
+			overlays.Cut()
```
```suggestion
			ClearOverlays()
```

LordNest committed:
Update code/modules/mob/living/carbon/human/inventory.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
delete mod

LordNest committed:
Update mods/fancy_sofas/_fancy_sofas.dme

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update maps/sierra/structures/other.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update maps/sierra/structures/other.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
len removal

Серьёзно, эти штуки стоят ровно в одном месте: у кадетов за стеклом. Пули никто не будет в космосе использовать там.
Шальные во время мерков/хейста не считаются

SuhEugene commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -21,6 +21,23 @@ This saves us from having to call add_fingerprint() any time something is put in
 				update_inv_l_hand(0)
 			else
 				update_inv_r_hand(0)
+			// [SIERRA-ADD] - SSINPUT
```
В README.md мода SSINPUT надо обозначить в графе кор кода это изменение

SuhEugene commented on maps/sierra/structures/other.dm:
```diff
@@ -10,3 +10,137 @@
+	var/list/bulletholes = list()
+
+/obj/item/target/use_tool(obj/item/W, mob/living/user, list/click_params)
+	. = ..()
+	if(isWelder(W))
+		var/obj/item/weldingtool/WT = W
+		if(WT.remove_fuel(0, user))
+			ClearOverlays()
+			bulletholes.Cut()
+			hp = initial(hp)
+			to_chat(usr, "<span class='notice'>You slice off [src]'s uneven chunks of aluminium and scorch marks.</span>")
+			return
```
Это пиздец.
Юз тул не так работает.

```suggestion
	if(isWelder(W))
		var/obj/item/weldingtool/WT = W
		if(WT.remove_fuel(0, user))
			ClearOverlays()
			bulletholes.Cut()
			hp = initial(hp)
			to_chat(usr, "<span class='notice'>You slice off [src]'s uneven chunks of aluminium and scorch marks.</span>")
		return TRUE
	return ..()
```

SuhEugene commented on maps/sierra/structures/other.dm:
```diff
@@ -10,3 +10,137 @@
+/obj/item/target/bullet_act(obj/item/projectile/Proj)
+	var/p_x = Proj.p_x + pick(0,0,0,0,0,-1,1) // really ugly way of coding "sometimes offset Proj.p_x!"
+	var/p_y = Proj.p_y + pick(0,0,0,0,0,-1,1)
+	var/decaltype = 1 // 1 - scorch, 2 - bullet
```
Это я щас сам перепишу

SuhEugene commented on maps/sierra/structures/other.dm:
```diff
@@ -10,3 +10,137 @@
+	var/b2y2 = 0
+
+/datum/bullethole/New(obj/item/target/Target, pixel_x = 0, pixel_y = 0)
+	if(!Target) return
```
Ёбаный рот, кто с заглавной переменные называет... Финики бля...

SuhEugene committed:
Rewrite targets shitcode

SuhEugene committed:
Merge branch 'dev-sierra' into LN-Sbay-OldNY

LordNest committed:
ридми

LordNest committed:
индентацию чиним

LordNest committed:
ЗАПЯТЫЕ

SuhEugene committed:
Всё по завещаниям модуляризации
